File: compare.py
import time
from timeit import default_timer

secret = "1715762622-Mb12K2Epbm46OhMOmzhrBOueEPxCMB"
len_secret = len(secret)
result = True

print("# (Changed character index) (time in nanosecond per comparison)")

for x in range(len_secret):
    input_str = list(secret)
    input_str[x] = '0'
    input_str = ''.join(input_str)

    start = default_timer()
    for _ in range(int(1e8)):
        result = input_str == secret
    end = default_timer()

    print(f"{x} {10 * (end - start):.9f}")

print(f"# result: {result}")

File: compare.php
<?php

$secret = "1715762622-Mb12K2Epbm46OhMOmzhrBOueEPxCMB";
$len = strlen($secret);
$result = 0;

echo "# (Changed character index) (time in nanosecond per comparison)\n";

for ($x = 0; $x < $len; $x++) {
    $input = $secret;
    $input[$x] = '0';

    $start = microtime(true);
    for ($i = 0; $i < 1e8; $i++) {
        $result ^= $input == $secret;
    }
    $end = microtime(true);

    $time = ($end - $start) * 10; // Convert to nanoseconds
    echo "$x $time\n";
}

echo "# result: $result\n";